thirdparty/faiss/faiss/utils/distances.cpp

```diff
@@ -211,30 +214,59 @@ void exhaustive_inner_product_seq_impl(
     using SingleResultHandler = typename BlockResultHandler::SingleResultHandler;
     int nt = std::min(int(nx), omp_get_max_threads());
 
+#ifdef FAISS_WITH_DNNL
+    if (is_dnnl_enabled()) {
+        float *res_arr = NULL;
+
+        comput_f32bf16f32_inner_product(nx, d, ny, d, const_cast<float*>(x), const_cast<float*>(y), &res_arr);
```

Comment:
Can we implement a dynamic hook like all other simd in Knowhere?

Reply:
We have also considered following the other simd interface, but due to the implementation of AMX, it may be a bit incompatible with the current interface:
1. AMX prefers batch data calculation, and it's library will schedule multiple threads on its own.
2. The return value is a array for batch data operation.
So if we use dynamic hook, maybe need add new interface for batch data operation, and call the new interface when AMX is available.

Reply:
@liliu-z We are planning to port code to adapt dynamic hook, do you have any other suggestions?